redshift/redshift.go

```diff
@@ -19,59 +17,55 @@ type dbExecCloser interface {
 	Exec(query string, args ...interface{}) (sql.Result, error)
 }
 
+// S3Info holds the information necessary to copy data from s3 buckets
+type S3Info struct {
+	Region    string
+	AccessID  string
+	SecretKey string
+}
+
 // Redshift wraps a dbExecCloser and can be used to perform operations on a redshift database.
 type Redshift struct {
 	dbExecCloser
-	accessID  string
-	secretKey string
+	s3Info S3Info
 }
 
 var (
```

Comment:
Ha, ok, though normally I might say it's a better practice to remove it entirely then add it back in. That confuses something like git-blame less!